### PR TITLE
Prepare for release 1.5

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
# nf-core/mhcquant pull request

Many thanks for contributing to nf-core/mhcquant!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, also make a PR on the [nf-core/mhcquant branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/mhcquant)
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Documentation in `docs` is updated
- [x] `CHANGELOG.md` is updated
- [x] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -17,12 +18,12 @@ def helpMessage() {
...
 
-    nextflow run nf-core/mhcquant --mzmls '*.mzML' --fasta '*.fasta' --vcf '*.vcf' --class_1_alleles 'alleles.tsv' --include_proteins_from_vcf --predict_class_1 --refine_fdr_on_predicted_subset -profile standard,docker
+    nextflow run nf-core/mhcquant --sample_sheet 'sample_sheet.tsv' --fasta 'SWISSPROT_2020.fasta'  --allele_sheet 'alleles.tsv'  --predict_class_1  --refine_fdr_on_predicted_subset -profile standard,docker
 
     Mandatory arguments:
-      --mzmls [file]                            Path to input data (must be surrounded with quotes)	
-      --raw_input [bool]                        Specify whether raw files should be used as input
-      --raw_files [file]                        Path to input raw data (must be surrounded with quotes)
+      --sample_sheet [file]                     Path to sample sheet (must be surrounded with quotes)
+      --raw_input [bool]                        Specify whether raw files are used as input
```
Comment (Zethson) [inline main.nf:25]: Would it make sense to merge raw_input and mzml_input to a single 'spectra' or as 'input' (longer discussion) and just identify the file format (raw or mzml) on the fly and use the appropriate entry point?

File: main.nf [hunk 2]
```diff
@@ -795,27 +847,29 @@ process run_percolator {
 
     script:
     if (params.description_correct_features > 0 && params.klammer){
+
     """
-    PercolatorAdapter -in ${id_file_psm} \\
-                       -out ${id_file_psm.baseName}_perc.idXML \\
+    OMP_NUM_THREADS=${task.cpus} PercolatorAdapter -in ${id_file_psm} \\
```
Comment (Zethson) [inline main.nf:852]: This sets an ENV variable right?
Maybe add a new line before calling Percolator to separate this more clearly?

File: nextflow.config [hunk 3]
```diff
@@ -12,6 +12,7 @@ params {
   name = false
   multiqc_config = "$baseDir/assets/multiqc_config.yaml"
   email = false
+  email_on_fail = false
```
Comment (Zethson) [inline nextflow.config:15]: Should this really be the default?

File: main.nf (hunk 1 above)
Comment (timosachsenberg) [reply main.nf:25]: yep that is what I also wanted to ask

File: main.nf [hunk 4]
```diff
@@ -94,14 +94,91 @@ if (params.help) {
...
 // Validate inputs
-if(!params.raw_input){
-   params.mzmls = params.mzmls ?: { log.error "No spectra data privided. Make sure you have used the '--mzmls' option."; exit 1 }()
+//MS Input
+if (params.sample_sheet)  {
+   sample_sheet = file(params.sample_sheet)
+
+   Channel.from( sample_sheet )
+                .splitCsv(header: true, sep:'\t')
+                .map { col -> tuple("${col.ID}", "${col.Sample}", "${col.Condition}", file("${col.ReplicateFileName}", checkifExists: true))}
```
Comment (timosachsenberg) [inline main.nf:103]: in the doc you use Samples but there it is col.Sample
I would make this consistent (and maybe prefer "Sample" over "Samples" as column header)

File: main.nf [hunk 5]
```diff
@@ -94,14 +94,91 @@ if (params.help) {
...
 params.outdir = params.outdir ?: { log.warn "No output directory provided. Will put the results into './results'"; return "./results" }()
 
+
+//Allele Input
+if (params.predict_class_1 || params.predict_class_2)  {
+   allele_sheet = file(params.allele_sheet, checkIfExists: true)
+
+   Channel.from( allele_sheet )
+                .splitCsv(header: true, sep:'\t')
+                .map { col -> tuple("${col.Sample}", "${col.HLA_Alleles_Class_1}", "${col.HLA_Alleles_Class_2}")}
```
Comment (timosachsenberg) [inline main.nf:157]: same here

File: main.nf [hunk 6]
```diff
@@ -185,50 +262,15 @@ variant_snp_filter="-fSNP"
...
-}
 
 /*
  * Create a channel for input fasta file
  */
 if( params.include_proteins_from_vcf) {
     Channel
         .fromPath( params.fasta )
+        .spread(ch_samples_for_fasta)
+        .flatMap{it -> [tuple(it[1],it[2],it[0])]}
```
Comment (timosachsenberg) [inline main.nf:273]: sadd comment what it points to

File: main.nf [hunk 7]
```diff
@@ -239,6 +281,8 @@ if( params.include_proteins_from_vcf) {
 } else if( params.skip_decoy_generation) {
     Channel
         .fromPath( params.fasta )
+        .spread(ch_samples_for_fasta)
+        .flatMap{it -> [tuple(it[1],it[2],it[0])]}
```
Comment (timosachsenberg) [inline main.nf:285]: same here

File: main.nf [hunk 8]
```diff
@@ -247,6 +291,8 @@ if( params.include_proteins_from_vcf) {
 } else {
     Channel
         .fromPath( params.fasta )
+        .spread(ch_samples_for_fasta)
+        .flatMap{it -> [tuple(it[1].toInteger(),it[2],it[0])]}
```
Comment (timosachsenberg) [inline main.nf:295]: here it 1 is converted to integer. in the other parts not. is that right

File: main.nf (hunk 1 above)
Comment (Leon-Bichmann) [reply main.nf:25]: ok yep I remember the discussion, I'll have another look at the proteomicsLFQ

File: nextflow.config (hunk 3 above)
Comment (Leon-Bichmann) [reply nextflow.config:15]: I thought, that's a special option for ppl who wanna set their emails when running the pipeline

File: main.nf (hunk 4 above)
Comment (Leon-Bichmann) [reply main.nf:103]: ok will change this

File: main.nf (hunk 8 above)
Comment (Leon-Bichmann) [reply main.nf:295]: ok right, have to check whether it is really necessary

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,17 @@
 # nf-core/mhcquant: Changelog
 
+## v1.5 nf-core/mhcquat "Flying Fish" - 2020/04/18
+
+### `Fixed`
+
+- raise OpenMS version to 2.5
+- adapt workflow accoringly with new options
+- integrate sample, allele and vcf instead of file dirs
```
Comment (lkuchenb) [inline CHANGELOG.md:9]: Sheets?

File: main.nf (hunk 6 above)
Comment (lkuchenb) [reply main.nf:273]: Why use `flatMap` + list encapsulation instead of just `map` + tuple?

File: main.nf [hunk 10]
```diff
@@ -94,14 +94,91 @@ if (params.help) {
...
+
+   Channel.from( sample_sheet )
+                .splitCsv(header: true, sep:'\t')
+                .map { col -> tuple("${col.ID}", "${col.Sample}", "${col.Condition}", file("${col.ReplicateFileName}", checkifExists: true))}
+                .flatMap{it -> [tuple(it[0],it[1].toString(),it[2],it[3])]}
+                .into { ch_samples_from_sheet; ch_samples_for_fasta}
+
+}
+
+if(params.mzml_input){
```
Comment (lkuchenb) [inline main.nf:109]: Having two flags `mzml_input` and `raw_input` seems unnecessary and bares the risk if the user specifying both, which is not caught as far as I can tell. I would suggest to make one behaviour the default, e.g. raw files, and add a flag `--from_mzml` which switches the behaviour.

File: CHANGELOG.md (hunk 9 above)
Comment (Leon-Bichmann) [reply CHANGELOG.md:9]: right sheets are missing

File: main.nf (hunk 10 above)
Comment (Leon-Bichmann) [reply main.nf:109]: ok, ill fix this in hands with the --spectra input comment from Lukas and Timo

File: main.nf (hunk 1 above)
Comment (Leon-Bichmann) [reply main.nf]: ok we changed it to input

File: main.nf (hunk 8 above)
Comment (Leon-Bichmann) [reply main.nf]: ok should be annotated now

File: main.nf (hunk 4 above)
Comment (Leon-Bichmann) [reply main.nf]: is fixed now

File: main.nf (hunk 2 above)
Comment (Leon-Bichmann) [reply main.nf]: solved

**Comment by Leon-Bichmann** [comment]
if everything else is fine, please approve and we'll try to make the release today

**Pull request merged**